evap/student/templates/student_vote.html

```diff
@@ -250,50 +250,44 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
+                const originalText = submitButton.innerText;
 
-            // Taken from http://stackoverflow.com/questions/9177252/detecting-a-redirect-in-jquery-ajax
-            // This hacks jQuery to give us this xhr object which is used below
-            var xhr;
-            var _orgAjax = jQuery.ajaxSettings.xhr;
-            jQuery.ajaxSettings.xhr = function () {
-              xhr = _orgAjax();
-              return xhr;
-            };
+                submitButton.innerText = "{% trans 'Submitting...' %}";
+                submitButton.disabled = true;
 
-            form.submit(function (event) {
-                event.preventDefault(); // don't use the default submission
-                var submitButton = $('#vote-submit-btn');
-                var originalText = submitButton.text();
-
-                submitButton.text("{% trans 'Submitting...' %}");
-                submitButton.prop('disabled', true);
-
-                $.ajax({
-                    type: form.attr('method'),
-                    url: form.attr('action'),
-                    data: form.serialize(),
-                    success: function (data, s, req) {
-                        if(data === "{{ success_magic_string }}") {
+                fetch(form.action, {
+                    body: new FormData(form),
+                    headers: CSRF_HEADERS,
+                    method: form.method,
+                }).then(response => {
+                    assert(response.ok);
+
+                    response.text().then((response_text) => {
+                        if(response_text === "{{ success_magic_string }}") {
                             sisyphus.manuallyReleaseData();
                             window.location.replace("{{ success_redirect_url }}");
                         } else {
                             // resubmit without this handler to show the site with the form errors
-                            form.unbind("submit");
-                            form.submit();
+                            form.removeEventListener("submit", submitListener);
+                            form.requestSubmit();
                         }
-                    },
-                    error: function(data) {
-                        // show a warning if the post isn't successful
-                        document.getElementById('submit-error-warning').style.display = 'block';
-                        submitButton.text(originalText);
-                        submitButton.prop('disabled', false);
-                    }
+                    });
+                }).catch(error => {
+                    // show a warning if the post isn't successful
+                    document.getElementById('submit-error-warning').style.display = 'block';
+                    submitButton.innerText = originalText;
+                    submitButton.disabled = false;
                 });
+
                 return false;
```

Comment:
Since we are using `preventDefault` above, I suppose we don't need this?

https://stackoverflow.com/a/1357151/13679671